jaxb/src/main/java/feign/jaxb/JAXBContextFactory.java

```diff
@@ -30,28 +35,38 @@
  */
 public final class JAXBContextFactory {
 
-  private final ConcurrentHashMap<Class, JAXBContext>
-      jaxbContexts =
-      new ConcurrentHashMap<Class, JAXBContext>(64);
+  private final ConcurrentHashMap<Integer, JAXBContext> jaxbContexts = 
+          new ConcurrentHashMap<Integer, JAXBContext>(64);
   private final Map<String, Object> properties;
+  private final CopyOnWriteArraySet<Class<?>> jaxbClasses =
+          new CopyOnWriteArraySet<Class<?>>();
 
   private JAXBContextFactory(Map<String, Object> properties) {
     this.properties = properties;
   }
 
+  private JAXBContextFactory(Map<String, Object> properties, Set<Class<?>> jaxbClasses) {
+      this.properties = properties;
+      this.jaxbClasses.addAll(jaxbClasses);
+  }
+  
   /**
    * Creates a new {@link javax.xml.bind.Unmarshaller} that handles the supplied class.
    */
-  public Unmarshaller createUnmarshaller(Class<?> clazz) throws JAXBException {
-    JAXBContext ctx = getContext(clazz);
+  public Unmarshaller createUnmarshaller(Class<?>... classes) throws JAXBException {
+    if (this.jaxbClasses.size() > 0)
+        classes = this.jaxbClasses.toArray(new Class<?>[0]);
+    JAXBContext ctx = getContext(classes);
     return ctx.createUnmarshaller();
   }
 
   /**
    * Creates a new {@link javax.xml.bind.Marshaller} that handles the supplied class.
    */
-  public Marshaller createMarshaller(Class<?> clazz) throws JAXBException {
-    JAXBContext ctx = getContext(clazz);
+  public Marshaller createMarshaller(Class<?>... classes) throws JAXBException {
+    if (this.jaxbClasses.size() > 0)
```

Comment:
Since you have this test in both `create` methods, could you move this test to `getContext` instead?